### ISSUE TITLE
[Feature Request] Wifi Tap Timer / Sprinkler System Support

### ISSUE THREAD

**Issue by tommymac2709** [rospogrigio/localtuya-homeassistant#582]
The integration discovers the wifi hub that connect to the Holman WX1 WiFi Tap Timer control, but fails to authenticate (all detailed are correct). Other Tuya integrations also fail to pick it up, so if support for these devices can be added, that'd be awesome!

**Comment by ganttview** [comment]
It's working on this fork: https://github.com/darek-margas/localtuya

See instructions at https://github.com/darek-margas/localtuya/blob/master/Holman%20WX1%20WiFi%20Irrigation%20Tap.md. 

Remember to plug in your device local_key at the appropriate spots. My device_id's (CID) were 03B for the hub socket and 031 for the irrigation controller.

**Comment by lawrencelnm** [comment]
Hi, I second the proposal for the Holman irrigation systems (running on Tuya) to be supported as part of this repository. As per previous comment, https://github.com/darek-margas/localtuya has got this working so hopefully it will be an easy merge of that code back into this branch.

**Comment by mgbernal** [comment]
Is any advance with this?

**Comment by fishy242** [comment]
Yeah, it has been done in other fork and looks like a easy merge. It is very useful so could consider merging it?

**Comment by lambsfry** [comment]
Adding more support for the merge

**Comment by jamesbannan** [comment]
Also adding support for the code merge. Just bought a few of these WX1 watering systems and would love to get them integrated properly.

**Comment by Et3rnal** [comment]
Any hope for this?
BTW K-RAIN devices (also use the Holman app, aka tuya)

**Comment by epajfl00** [comment]
Agree, should merge required changes for CID sub parameter to be added (eg for Holman WX1 tap timer).
In the meantime, in order to use a forked version from "https://github.com/darek-margas/localtuya" or "https://github.com/NeriedAU/localtuya" do you overwrite the existing version from "https://github.com/rospogrigio/localtuya" or can add the other version to the custom_components folder (with a different name), so 2 versions are running at the same time?

**Comment by Et3rnal** [comment]
With the latest update, I was able to add k-rain without a need for a new fork and here is my documented process in case anymore end up here or you want to include it in an upcoming release

https://community.home-assistant.io/t/k-rain-wi-fi-irrigation-controller-configration-for-local-tuya/525266/3

**Comment by CloCkWeRX** [comment]
Cross linking other conversations here, which include a number of sample DPs particularly for the australian version(s).
https://community.home-assistant.io/t/holman-wifi-tap-timers-intergration/313920/67

**Comment by xXAzazelXx** [comment]
Any updates on this?

**Comment by Bungisard** [comment]
Hi there, I see the Tuya integration detects the Holman Wireless Hub and WX1 tap automatically now but they are unsupported. Is anything being done to support these in the future??
